From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: William Douglas <william.douglas@intel.com>
Date: Wed, 4 Dec 2019 11:03:58 -0800
Subject: [PATCH] Add option for systemd-nspawn to prevent stdin being a ptty

Due to stdin (and stdout/stderr) being ptty's by default without the
--console=pipe option to systemd-nspawn, builds will assume they are
interactive and sometimes hang.

Rebased for mock 2.4 update by Francisco Boni <boboniboni@gmail.com>.
---
 mock/py/mockbuild/util.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mock/py/mockbuild/util.py b/mock/py/mockbuild/util.py
index cabe53b..7eede3b 100644
--- a/mock/py/mockbuild/util.py
+++ b/mock/py/mockbuild/util.py
@@ -1034,7 +1034,7 @@ def setup_default_config_opts(unprivUid, version, pkgpythondir, unprivGid=None):
     config_opts['isolation'] = None
     config_opts['use_nspawn'] = None
     config_opts['rpmbuild_networking'] = False
-    config_opts['nspawn_args'] = ['--capability=cap_ipc_lock']
+    config_opts['nspawn_args'] = ['--capability=cap_ipc_lock', '--console=pipe']
     config_opts['use_container_host_hostname'] = True
 
     config_opts['use_bootstrap'] = True
